src/sparsezoo/v2/objects/model_directory.py

```diff
@@ -238,9 +237,20 @@ def download(self, directory_path: str, override: bool = False) -> bool:
             )
         else:
             downloads = []
-            for file in self.files:
-                downloads.append(self._download(file, directory_path))
-
+            for key, file in self._files_dictionary.items():
+                if file is not None:
+                    downloads.append(self._download(file, directory_path))
+                else:
+                    if strict_mode:
+                        raise ValueError(
+                            f"Attempted to download file {key}, " f"but it is `None`!"
+                        )
+
+                    else:
+                        logging.warning(
+                            f"Attempted to download file {key}, "
+                            f"but it is `None`. The file is being skipped..."
+                        )
         return all(downloads)
```

Comment:
nit: again, can be ignored, but do we really need to create a `downloads` list in this case?

Reply:
I still like it being there for the debug properties, to be honest.